### DISCUSSION TITLE
MESH files from pyvista UnstructuredGrid

### DISCUSSION THREAD

**Discussion by wangjx520** [jtecklenburg/voromesh#3] [General]
1. I would like to know if it is possible to generate MESH file directly using UnstructuredGrid objects from pyvista? 2. After stretching the UnstructuredGrid can I further modify the top and bottom levels of certain grids to get unequal thickness layers?

**Comment by jtecklenburg** [comment]
Try toughio [1]. This library uses meshio to read meshes and can write TOUGH meshes. Meshio [2] can read a lot of different mesh formats and VTU as well.

[1] https://keurfonluu.github.io/toughio/
[2] https://github.com/nschloe/meshio

**Comment by jtecklenburg** [comment]
Voromesh does not support a spatial changing layer thickness.
